addons/godot-xr-tools/objects/snap_zone.gd

```diff
@@ -208,3 +250,30 @@ func _set_grab_distance(new_value: float) -> void:
 	grab_distance = new_value
 	if is_inside_tree() and $CollisionShape:
 		$CollisionShape.shape.radius = grab_distance
+
+
+# Called when the snap mode property has been modified
+func _set_snap_mode(new_value: int) -> void:
+	snap_mode = new_value
+	if is_inside_tree():
+		_update_snap_mode()
+
+
+# Handle changes to the snap mode
+func _update_snap_mode() -> void:
+	set_process(snap_mode == SnapMode.RANGE)
```

Comment:
We should loop through the objects currently in range and either connect or disconnect our dropped signal when applicable.

If someone changes the snap mode in runtime we could end up with it not working properly

Reply:
Updated to handle dynamic switching of modes correctly.